src/config/backend/EdFi.DmsConfigurationService.Backend.Postgresql.Test.Integration/ApplicationTests.cs

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Numerics;
+using EdFi.DmsConfigurationService.DataModel;
+using FluentAssertions;
+
+namespace EdFi.DmsConfigurationService.Backend.Postgresql.Test.Integration;
+
+public class ApplicationTests : DatabaseTest
+{
+    private readonly IRepository<Application> _repository = new ApplicationRepository(
+        Configuration.DatabaseOptions
+    );
+    private long _vendorId;
+
+    [TestFixture]
+    public class InsertTest : ApplicationTests
```

Comment:
Maybe add a `public class InsertFailureTest` after this that fails to insert due to the vendor id not existing. You might have to make a new `InsertResult.InsertFailureReferenceNotFound` result